lib/solveEquation.js

```diff
@@ -112,45 +112,91 @@ function step(equation, symbolName) {
+    step.changeType = MathChangeTypes.SIMPLIFY_LEFT_SIDE;
+    steps = addStep(steps, step, debug);
+  }
+  else if (leftSubSteps.length > 1) {
+    const lastStep = leftSubSteps[leftSubSteps.length - 1];
+    const finalEquation = EquationStatus.resetChangeGroups(lastStep.newEquation);
+    // no change groups are set here - too much is changing for it to be useful
+    const simplifyStatus = new EquationStatus(
+      MathChangeTypes.SIMPLIFY_LEFT_SIDE,
+      oldEquation, finalEquation, leftSubSteps);
+    steps = addStep(steps, simplifyStatus, debug);
   }
 
+  // update `equation` to have the new simplified left node
+  if (steps.length > 0) {
+    equation = EquationStatus.resetChangeGroups(
+      steps[steps.length - 1 ].newEquation);
+  }
+
+  // the updated equation from simplifing the left side is the old equation
+  // (ie the "before" of the before and after) for simplifying the right side.
+  oldEquation = equation.clone();
+
   let rightSteps = stepThrough(equation.rightNode, false, firstTime);
+  let rightSubSteps = [];
   for (let i = 0; i < rightSteps.length; i++) {
     let step = rightSteps[i];
-    equation.rightNode = math.parse(step.asciimath);
-    equationStatus = new EquationStatus(equation, step.changeType);
-    steps = addStep(steps, equationStatus, debug);
+    rightSubSteps.push(EquationStatus.addRightStep(equation, step));
+  }
+  if (rightSubSteps.length === 1) {
+    const step = rightSubSteps[0];
+    step.changeType = MathChangeTypes.SIMPLIFY_RIGHT_SIDE;
+    steps = addStep(steps, step, debug);
+  }
+  else if (rightSubSteps.length > 1) {
+    const lastStep = rightSubSteps[rightSubSteps.length - 1];
+    const finalEquation = EquationStatus.resetChangeGroups(lastStep.newEquation);
+    // no change groups are set here - too much is changing for it to be useful
+    const simplifyStatus = new EquationStatus(
+     MathChangeTypes.SIMPLIFY_RIGHT_SIDE,
+     oldEquation, finalEquation, rightSubSteps);
+    steps = addStep(steps, simplifyStatus, debug);
   }
 
   return steps;
 }
 
 // adds a step to the given steps based on the given equationStatus
 function addStep(steps, equationStatus, debug) {
-  steps.push({
-    'changeType': equationStatus.changeType,
-    'asciimath': equationStatus.equation.print(),
-    'equation': equationStatus.equation,
-  });
   if (debug) {
     // eslint-disable-next-line
-    console.log('\n' + explanation);
+    console.log('\n' + equationStatus.changeType);
     // eslint-disable-next-line
-    console.log(asciimath);
+    console.log(equationStatus.newEquation.print());
+    if (equationStatus.substeps.length > 0) {
+      console.log("\n substeps: ");
+    }
   }
+  let substeps = [];
+  equationStatus.substeps.forEach(substep => {
+    substeps = addStep(substeps, substep, debug);
+  });
+  steps.push({
+    'changeType': equationStatus.changeType,
+    'asciimath': equationStatus.newEquation.print(),
+    'newEquation': equationStatus.newEquation,
```

Comment:
yes! I think we do want the oldNode/oldEquation for the custom changetypes that say like "add __ and __" where we'd only know the blanks from the before node/equation